src/main/java/io/smartcat/ranger/core/parser/ValueExpressionParser.java

```diff
@@ -385,8 +421,11 @@ public Rule rangeValueDouble() {
      * @return Long range value definition rule.
      */
     public Rule rangeValueLong() {
-        return Sequence(function("random", Sequence(longLiteral(), "..", longLiteral())),
-                push(new RangeValueLong((Long) pop(1), (Long) pop())));
+        return Sequence(
+                function("random", Sequence(longLiteral(), "..", longLiteral(), Optional(comma(), distribution()))),
+                push(peek() instanceof Distribution
+                        ? new RangeValueLong((Long) pop(2), (Long) pop(1), (Distribution) pop())
+                        : new RangeValueLong((Long) pop(1), (Long) pop())));
```

Comment:
I think I would go for readability here and wrap this instanceof part in private functions, I know there would be 3 of them but it will be easier to reason on different levels of abstraction.

```
return Sequence(function("random", Sequence(longLiteral(), "..", longLiteral(), Optional(comma(), distribution()))), push(longValue());
```

And
```
private RangeValueLong longValue() {
   return peek() instanceof Distribution
                         ? new RangeValueLong((Long) pop(2), (Long) pop(1), (Distribution) pop())
                         : new RangeValueLong((Long) pop(1), (Long) pop());
}
```